### ISSUE TITLE
select a (or a few of) tracer(s) that has specific x, y, and z coordinates

### ISSUE THREAD

**Issue by wenrongcao** [JuliaGeodynamics/LaMEM.jl#60]
I ran the following example code without issues except the need to update some older syntax (see my last line of comment):
```
using LaMEM, GeophysicalModelGenerator, Plots
model  = Model(Grid(nel=(16,16,16), x=[-1,1], y=[-1,1], z=[-1,1]), PassiveTracers(Passive_Tracer=1, PassiveTracer_Box=[-1,1,-1,1,-1,1]))
matrix = Phase(ID=0,Name="matrix",eta=1e20,rho=3000)
sphere = Phase(ID=1,Name="sphere",eta=1e23,rho=3200)
add_phase!(model, sphere, matrix)
add_sphere!(model,cen=(0.0,0.0,0.0), radius=(0.5, ))
run_lamem(model,1)
```
I am trying to select a (or a few of) tracer that has specific x, y, and z coordinates because the index (ID) of the tracer does not bear explicit spatial meanings.  Given the CartData structure, I tried to select tracers, for example, x> 0, by doing:

`data,time = read_LaMEM_timestep(model, 0, passive_tracers=true)`
`ID = findall(data.x .> 0)` 

Yet, the error says: 
`MethodError: no method matching is less(::Int64, ::Unitful.FreeUnits{(km,), 𝐋, nothing})`

It seems more information is needed about the unit used.  

I see a workaround that is to specify a small tracer area at the beginning of the code instead of having a wide distribution of tracers everywhere. But it will be nice to be able to select a tracer by designating the x,y,z range of interest. 

---------------------
BTW, some syntax in this example needs to be updated: Read_LaMEM_timestep -- > read_LaMEM_timestep and PassiveTracer_Time --> passivetracer_Time.

_Originally posted by @wenrongcao in https://github.com/JuliaGeodynamics/LaMEM.jl/issues/35#issuecomment-2311779305_

**Comment by wenrongcao** [comment]
I figured out how to do it: data.x is a structure containing three fields: value, unit, and isdimensional. I need to extract the value:
`ID = findall(x -> x > 0, data.x.val)` 
It returns all tracer ID whose x >0. 

If I want to select tracers having x>0 and z>0, I need to do:
`ID = findall(i -> data.x.val[i] > 0 && data.z.val[i] > 0, 1:length(data.x.val))`

**Comment by boriskaus** [comment]
yes this is how this should be done. Note that you can do this as well in Julia:
```Julia
ID = findall(data.x.val .> 0)
```
Note the dot in front of `>`, which implies that it is applied to every point the array `data.x.val`.
Likewise, you can write:
```Julia
ID = findall(data.x.val .> 0 .&& data.z.val .> 0)
``` 
which is slightly more compact than what you wrote.

**Comment by wenrongcao** [comment]
Thanks! The method also works. Will be good if such an example can be included in the user guide: [https://juliageodynamics.github.io/LaMEM.jl/dev/](url)  in the future.

**Comment by boriskaus** [comment]
Why don't you prepare the example and make a pull request? That would really help to improve the code. 
Same with the typos you noticed above.

**Comment by wenrongcao** [comment]
Will do!

**Comment by wenrongcao** [comment]
This issue can be closed with the pull request #61 .

**Comment by wenrongcao** [comment]
This issue can be closed with the pull request #61 .
